src/subscriptions.ts

```diff
@@ -107,12 +126,26 @@ export async function getSubscribedTransactions(
           `Catching up from round ${startRound} to round ${algodSyncFromRoundNumber - 1} via indexer; this may take a few seconds`,
         )
 
+        // Retrieve and process transactions from indexer
         catchupTransactions = (
-          await algokit.searchTransactions(indexer, indexerPreFilter(filter, startRound, algodSyncFromRoundNumber - 1))
-        ).transactions
-          .flatMap((t) => getFilteredIndexerTransactions(t, filter))
-          .filter(indexerPostFilter(filter, arc28Events, subscription.arc28Events ?? []))
+          await Promise.all(
```

Comment:
This will result in parallel execution of filter requests. Should we chunk this with a max size, so if someone specifies a large number of filters, we don't hit requests limits for the node?